tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -3392,3 +3393,305 @@ def call(self, inputs, state):
 
     new_state = rnn_cell_impl.LSTMStateTuple(new_c, new_h)
     return new_h, new_state
+
+NTMControllerState = collections.namedtuple('NTMControllerState',
+      ('controller_state', 'read_vector_list', 'w_list', 'M', 'time'))
+
+class NTMCell(rnn_cell_impl.LayerRNNCell):
+  """Neural Turing Machine Cell with RNN controller.
+
+    Implementation based on:
+    https://arxiv.org/abs/1807.08518
+    Mark Collier, Joeran Beel
+
+    which is in turn based on the source code of:
+    https://github.com/snowkylin/ntm
+
+    and of course the original NTM paper:
+    Neural Turing Machines
+    https://arxiv.org/abs/1410.5401
+    A Graves, G Wayne, I Danihelka
+  """
+  def __init__(self, controller, memory_size,
+    memory_vector_dim, read_head_num, write_head_num, shift_range=1,
+    output_dim=None, clip_value=20, dtype=dtypes.float32,
+    reuse=None, name=None):
```

Comment:
maybe remove the reuse argument; i think we want to avoid having users deal w/ variable scopes moving forward.